jetbrains-core/it-211+/software/aws/toolkits/jetbrains/services/ecr/EcrPushIntegrationTest.kt

```diff
@@ -51,21 +59,56 @@ class EcrPushIntegrationTest {
 
         val project = projectRule.project
         runBlocking {
-            val serverInstance = getDockerServerRuntimeInstance()
+            val serverInstance = getDockerServerRuntimeInstance().runtimeInstance
             val ecrLogin = ecrClient.authorizationToken.authorizationData().first().getDockerLogin()
             val runtime = RemoteDockerApplicationRuntime.createWithPullImage(
                 RemoteDockerRuntime.create(DockerCloudConfiguration.createDefault(), project),
                 DockerAgentBuildImageConfig(System.currentTimeMillis().toString(), dockerfile, false)
             )
             // gross transform because we only have the short SHA right now
             val localImageId = runtime.agent.getImages(null).first { it.imageId.startsWith("sha256:${runtime.agentApplication.imageId}") }.imageId
-            val pushRequest = EcrPushRequest(
+            val pushRequest = ImageEcrPushRequest(
+                dockerServerRuntime = serverInstance,
                 localImageId = localImageId,
                 remoteRepo = remoteRepo,
                 remoteTag = remoteTag
             )
 
-            pushImage(projectRule.project, serverInstance, ecrLogin, pushRequest)
+            pushImage(projectRule.project, ecrLogin, pushRequest)
+        }
+
+        assertThat(
+            ecrClient.describeImages {
+                it.repositoryName(remoteRepo.repositoryName)
+            }.imageDetails()
+        ).anySatisfy {
+            assertThat(it.imageTags()).contains(remoteTag)
+            // would check the digest matches what we've uploaded, but it appears to change upon upload to ECR
+        }
+    }
+
+    @Test
+    fun testPushFromDockerfile() {
+        val remoteTag = UUID.randomUUID().toString()
+
+        val dockerfile = folder.newFile()
+        dockerfile.writeText(
+            """
+                # arbitrary base image with a shell
+                FROM public.ecr.aws/lambda/provided:latest
+                RUN touch $(date +%s)
+            """.trimIndent()
+        )
+
+        val ecrLogin = ecrClient.authorizationToken.authorizationData().first().getDockerLogin()
+        val config = dockerRunConfigurationFromPath(projectRule.project, remoteTag, dockerfile.absolutePath)
+        val pushRequest = DockerfileEcrPushRequest(
+            config.configuration as DockerRunConfiguration,
+            remoteRepo,
+            remoteTag
+        )
+        runBlocking {
+            pushImage(projectRule.project, ecrLogin, pushRequest)
```

Comment:
nit: I think we can remove the comment in the assert below, we are already testing it above with a different path